File: Boost-Python/MyTestClass/test_class.py
from TestClass import MyStack

############################################
print "Using vector constructor..."

stack = MyStack()

stack.push(3.2)
stack.push(3.1)
stack.push(3)

while not stack.empty():
    print stack.pop()
###########################################

###########################################
print "Using vector constructor..."

lst = [1,2,1.3]
stack.set(lst)

if stack.empty():
    print "Stack is already empty. Won't enter loop"
else:
    print "Stack isn't empty. Popping values."
while not stack.empty():
    print stack.pop()
###########################################

File: Boost-Python/MyTestClass/TestClass.cpp
#include <vector>

#include "TestClass.h"

MyStack::MyStack() { }

MyStack::MyStack(std::vector<double> _values)
{
  values = _values;
}

void MyStack::set(std::vector<double> newValues)
{
  values = newValues;
}

double MyStack::top()
{
  return values.back();
}

void MyStack::push(double value)
{
  values.push_back(value);
}

double MyStack::pop()
{
  double top = this->top();
  values.pop_back();
  return top;
}

bool MyStack::empty()
{
  return values.empty();
}

int MyStack::size() { return values.size(); }
